torchio/transforms/augmentation/intensity/random_simulation.py

```diff
@@ -0,0 +1,172 @@
+        >>> colin = Colin27(2008)
+        >>> # Using the default coefficients
+        >>> transform = RandomSimulation(label_keys='cls')
+        >>> # Using custom coefficients
+        >>> label_values = colin['cls'][DATA].unique()
+        >>> coefficients = {
+        >>>     i: {
+        >>>         'mean': i / len(label_values), 'std': 0.01
+        >>>     } for i in range(1, len(label_values))
+        >>> }
+        >>> transform = RandomSimulation(label_keys='cls', coefficients=coefficients)
+        >>> # Inpainting the simulated image on the original T1 image
+        >>> rescale_transform = RescaleIntensity((0, 1), (1, 99))   # Rescale intensity before inpainting
+        >>> simulation_transform = RandomSimulation(label_keys='cls', image_key='t1')
+        >>> transform = Compose([rescale_transform, simulation_transform])
+        >>> transformed = transform(colin)  # colin has a new key 'image' with the simulated image
+    """
+    def __init__(
+            self,
+            label_keys: Optional[Union[str, List[str], Tuple[str, ...]]] = None,
+            coefficients: Optional[Dict[Union[str, TypeNumber], Dict[str, TypeRangeFloat]]] = None,
+            image_key: str = 'image',
+            binary: bool = True,
+            p: float = 1,
+            seed: Optional[int] = None,
+            **kwargs
+            ):
+        super().__init__(p=p, seed=seed, **kwargs)
+        self.label_keys = label_keys
+        self.coefficients = self.parse_coefficients(coefficients)
+        self.image_key = image_key
+        self.binary = binary
+
+    def apply_transform(self, sample: Subject) -> dict:
+        random_parameters_images_dict = {}
+        original_image = sample.get(self.image_key)
+        final_image = None
+
+        if isinstance(self.label_keys, str):
+            image_dict = sample[self.label_keys]
+            data = image_dict[DATA]
+            labels = data.unique()[1:]
+            for i, label in enumerate(labels, 1):
+                mean, std = self.get_params(i)
+                random_parameters_images_dict[i] = {'mean': mean, 'std': std}
+                tissue = self.generate_tissue(data == label, mean, std)
+                if final_image is None:
+                    final_image = Image(type=INTENSITY, affine=image_dict[AFFINE], tensor=tissue[0])
+                else:
+                    final_image[DATA] += tissue
+
+        else:
+            for image_name, image_dict in sample.get_images_dict(intensity_only=False).items():
+                if image_dict[TYPE] == LABEL:
+                    if self.label_keys is None or image_name in self.label_keys:
+                        mean, std = self.get_params(image_name)
+                        random_parameters_images_dict[image_name] = {'mean': mean, 'std': std}
+                        if self.binary:
+                            data = image_dict[DATA] >= 0.5          # May fail if PV between more than 2 structures
```

Comment:
PV is partial volume? Why would it fail?

Reply:
because you can have 0.333 in each class, so the pixel will get no label ..
not sure if we need to correct here ... we can live with 0 pixel, -> they will be interpreted as belongin to the backgroud mask
on the other hand if a pixel belong to multiple class, it can be ok, just check than the simulated signal in this case is the mean of the 2 class (and not the sum)

Reply:
The code won't fail but as stated by @romainVala we could have errors in the binarization if a voxel belongs to more than 2 classes.

> on the other hand if a pixel belong to multiple class, it can be ok, just check than the simulated signal in this case is the mean of the 2 class (and not the sum)

I need to think about this...

Reply:
I'd just create a single label map from the probability maps using `argmax`.